Listening on control socket fails

I have  upgraded portable (1:12.0-1 -> 1:13.2-1) recently,then my wechat can't launch.This is the debug information from wechat:
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x55e92efb6e00]

goroutine 75 [running]:
main.watchSignalSocket(0xc00002a180)
        top.kimiblock.portable/portable-daemon/daemon.go:1221 +0x320
created by main.main in goroutine 1
        top.kimiblock.portable/portable-daemon/daemon.go:2787 +0x85b

--- a/lib/daemon/daemon.go
+++ b/lib/daemon/daemon.go
@@ -1214,6 +1214,7 @@ func watchSignalSocket(readyChan chan int8) {
 	socket, listenErr := net.Listen("unix", signalSocketPath)
 	if listenErr != nil {
 		pecho("crit", "Unable to listen for signal: " + listenErr.Error())
+		return
 	}
 	readyChan <- 1
 	pecho("debug", "Accepting signals")

